### ISSUE TITLE
bug fix for tags

### ISSUE THREAD

**Issue by richrobber2** [cyrusfirheir/twee3-language-tools#115]
bug description:
the bug seems to be caused when the tag has spaces so I say change it to some other text

This is for the "story map"
I believe this was a bug that only happens when the first tag is used with spaces

changed from
```js
  addPassageTag(passages: Passage[], tag: string) {
    for (const passage of passages) {
      if (!passage.tags.some((oldTag) => !oldTag.toLowerCase().localeCompare(tag.trim().toLowerCase()))){
        passage.tags.push(tag.trim());
      }
    }
  }
```

to
  ```js
addPassageTag(passages: Passage[], tag: string) {
    for (const passage of passages) {
      // parse for and remove spaces because spaces break the tags change it to a "_"
      if (!passage.tags.some((oldTag) => !oldTag.toLowerCase().localeCompare(tag.trim().replace(/\s/g, '_').toLowerCase()))){
        passage.tags.push(tag.trim().replace(/\s/g, '_'));
      }
    }
  }
```

this only changes it so that it uses regex to remove the middle spaces that cause bugs

**Comment by richrobber2** [comment]
this did not fix the bug im gonna look elsewhere dont even bother with this

**Comment by cyrusfirheir** [comment]
Should be fixed in v0.18.3.
